controllers/topology/cluster_controller_test.go

```diff
@@ -0,0 +1,377 @@
+		// Create spec fake setting to assert that template spec is non-empty for tests.
+		WithSpecFields(map[string]interface{}{"spec.template.spec.fakeSetting": true}).
+		Build()
+	controlPlaneTemplate := builder.ControlPlaneTemplate(ns.Name, "cp1").
+		WithInfrastructureMachineTemplate(infrastructureMachineTemplate).
+		Build()
+	bootstrapTemplate := builder.BootstrapTemplate(ns.Name, "bootstraptemplate").Build()
+
+	// 2) ClusterClass definitions including definitions of MachineDeploymentClasses used inside the ClusterClass.
+	machineDeploymentClass := builder.MachineDeploymentClass(ns.Name, "md1").
+		WithClass(workerClassName).
+		WithInfrastructureTemplate(infrastructureMachineTemplate).
+		WithBootstrapTemplate(bootstrapTemplate).
+		WithLabels(map[string]string{"foo": "bar"}).
+		WithAnnotations(map[string]string{"foo": "bar"}).
+		Build()
+	machineDeploymentClass2 := builder.MachineDeploymentClass(ns.Name, "md1").
+		WithClass(workerClassName2).
+		WithInfrastructureTemplate(infrastructureMachineTemplate).
+		WithBootstrapTemplate(bootstrapTemplate).
+		Build()
+	clusterClass := builder.ClusterClass(ns.Name, clusterClassName).
+		WithInfrastructureClusterTemplate(infrastructureClusterTemplate).
+		WithControlPlaneTemplate(controlPlaneTemplate).
+		WithControlPlaneInfrastructureMachineTemplate(infrastructureMachineTemplate).
+		WithWorkerMachineDeploymentClasses([]clusterv1.MachineDeploymentClass{*machineDeploymentClass, *machineDeploymentClass2}).
+		Build()
+
+	// 3) A Cluster object including a Cluster Topology Object and the MachineDeploymentTopology objects used in the
+	// Cluster Topology.
+	machineDeploymentTopology := builder.MachineDeploymentTopology("mdm").
+		WithClass(workerClassName).
+		WithReplicas(3).
+		Build()
+	machineDeploymentTopology2 := builder.MachineDeploymentTopology("mdm2").
+		WithClass(workerClassName).
+		WithReplicas(1).
+		Build()
+	cluster := builder.Cluster(ns.Name, clusterName).
+		WithTopology(
+			builder.ClusterTopology().
+				WithClass(clusterClass.Name).
+				WithMachineDeployment(machineDeploymentTopology).
+				WithMachineDeployment(machineDeploymentTopology2).
+				WithVersion("1.22.2").
+				WithControlPlaneReplicas(3).
+				Build()).
+		Build()
+
+	// Create a set of initObjects from the objects above to add to the API server when the test environment starts.
+	initObjs := []client.Object{
+		clusterClass,
+		cluster,
+		infrastructureClusterTemplate,
+		infrastructureMachineTemplate,
+		bootstrapTemplate,
+		controlPlaneTemplate,
+	}
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+
+	for _, obj := range initObjs {
+		err := env.Create(ctx, obj)
+		g.Expect(err).ToNot(HaveOccurred())
+	}
+	defer func() {
+		for _, obj := range initObjs {
+			err := env.Delete(ctx, obj)
+			g.Expect(err).ToNot(HaveOccurred())
+		}
+	}()
+	actualCluster := &clusterv1.Cluster{}
```

Comment:
```suggestion
	}()

	actualCluster := &clusterv1.Cluster{}
```
nit: I think an empty line to separate logical code blocks would be good here